models/junos/interfaces/junos_interfaces.yml

```diff
@@ -0,0 +1,99 @@
+---
+generator_version: 1.0
+metadata:
+  vesion: 1.1
+  status:
+  - preview
+  supported_by:
+  - 'Ansible Network'
+  copyright_str: Copyright 2019 Red Hat
+  license: gpl-3.0.txt
+info:
+  network_os: junos
+  resource: interfaces
+  version_added: 2.9
+  short_description: "This module provides configuration management of Interfaces on Juniper JUNOS network devices."
+  description:
+  - "Manage Interface on Juniper JUNOS network devices."
+  authors:
+  - Ganesh Nalawade (@ganeshrn)
+  notes:
+    - Tested against vSRX JUNOS version 17.3R1.10.
+schema:
+  type: object
+  description: The schema used for the argspec and docstring
+  properties:
+    config:
+      type: array
+      description: The provided configuration
+      items:
+        $ref: '#/definitions/config_dict'
+    state:
+      $ref: '#/definitions/state'
+
+definitions:
+  config_dict:
+    description:
+    - Specifies interfaces related configurations.
+    type: object
+    properties:
+      name:
+        description:
+        - The complete name of the interface to be managed, the interface name
+          should be in same format as seen in the running configurtion.
+        type: str
+        required: true
+      description:
+        description:
+        - This option represnets a text description of the interface.
+        type: str
+      enable:
+        description:
+        - This option takes a boolean value which controls state of the interface.
+          If value is C(True) the inerfaces is enabled, if value is C(False) interface
+          is disabled.
+        type: bool
+        default: true
+      speed:
+        description:
+        - This option configures the link speed for the given interface I(name).
```

Comment:
```suggestion
        - Interface link speed. Applicable for Ethernet interfaces only.
```

Reply:
are there a fixed set of speeds this can be set to? other models have something like:
type: str
enum: ['auto', '10', '100', '1000', '2500', '10000']

Reply:
speed varies based on the series of the JUNOS device, hence I have kept it as str